network/src/network.rs

```diff
@@ -91,6 +91,12 @@ impl NetworkState {
             .listen_addresses
             .iter()
             .chain(config.public_addresses.iter())
+            .filter(|addr| extract_peer_id(addr).is_some())
```

Comment:
user can set a public address without peer id in in the previous version:
```
[network]
listen_addresses = ["/ip4/0.0.0.0/tcp/8115"]
### Specify the public and routable network addresses
public_addresses = ["/ip4/179.46.242.43/tcp/8115"]
```

`local_node_info` rpc will return this address and appending the peer id to it:
```
curl -X POST "http://127.0.0.1:8114/" -d '{"jsonrpc":"2.0","method":"local_node_info","params":[],"id":1}' -H 'content-type: application/json'
{"jsonrpc":"2.0","result":{"active":true,"addresses":[{"address":"/ip4/179.46.242.43/tcp/8115/p2p/QmdaLwbqU7QRrZhUPD88y49dJUg3cquMwDXS6MnjEZpbMw","score":"0x1"},{"address":"/ip4/0.0.0.0/tcp/8115/p2p/QmdaLwbqU7QRrZhUPD88y49dJUg3cquMwDXS6MnjEZpbMw","score":"0x1"}],"connections":"0x4","node_id":"QmdaLwbqU7QRrZhUPD88y49dJUg3cquMwDXS6MnjEZpbMw","protocols":[{"id":"0x64","name":"/ckb/syn","support_versions":["1"]},{"id":"0x65","name":"/ckb/rel","support_versions":["1"]},{"id":"0x66","name":"/ckb/tim","support_versions":["1"]},{"id":"0x6e","name":"/ckb/alt","support_versions":["1"]},{"id":"0x3","name":"/ckb/flr","support_versions":["0.0.1"]},{"id":"0x4","name":"/ckb/disconnectmsg","support_versions":["0.0.1"]},{"id":"0x0","name":"/ckb/ping","support_versions":["0.0.1"]},{"id":"0x1","name":"/ckb/discovery","support_versions":["0.0.1"]},{"id":"0x2","name":"/ckb/identify","support_versions":["0.0.1"]}],"version":"0.40.0-pre (6e1092e 2021-01-29)"},"id":1}
```

it looks like this PR breaks this configration?

Reply:
Changed to add if not, so there will be no problems